ext/RMagick/rmimage.c

```diff
@@ -10441,7 +11487,18 @@ Image_random_threshold_channel(int argc, VALUE *argv, VALUE self)
 
     exception = AcquireExceptionInfo();
 
+#if defined(IMAGEMAGICK_7)
+    BEGIN_CHANNEL_MASK(new_image, channels);
+    {
```

Comment:
They are there to make sure that `GeometryInfo geometry_info;` is defined in a separate block and this is the style we use inside ImageMagick. And added the block because we really only need this for IM7.

Reply:
My comment got lost. These braces are here because `GeometryInfo geometry_info;` is not used by the other method. But it is probably better to move that up and in a `if defined(IMAGEMAGICK_7)` block. I will let you decide?

Reply:
I placed it in the braces so the variables are not declared in the middle of the code but at the start of a scope (code within braces). I could move the declartion up near `ExceptionInfo *exception;` and remove the braces but I would need another `if defined(IMAGEMAGICK_7)` block there then. You decide?

Reply:
I see. I think it's fine to leave it like this.

Reply:
I'm fine with moving it up, too, if you prefer.

Reply:
Will just leave it like this.